cloudevents/working-drafts/xml-format.md

```diff
@@ -0,0 +1,277 @@
+
+Preservation of type information for CloudEvent attributes is supported allowing
+custom extensions to be communicated without type loss.
+
+A schema-less approach has been taken favoring convention over rigid document
+structure.
+
+The namespace `http://cloudevents.io/xmlformat/v1` SHOULD be used, when
+namespace prefixes are used a prefix of `ce` is preferred but MUST NOT be expected
+from an XML document processing perspective.
+
+An XML document preamble SHOULD be included to ensure deterministic processing.
+
+## 2. Attributes
+
+The CloudEvents type system is mapped to the XML schema types as follows :
+
+| CloudEvents   |  XML Schema Type | Notes |
+| :-------------| :--------------- | :---- |
+| Boolean       | [xs:boolean][xml-primitives] | |
+| Integer       | [xs:int][xml-primitives] | |
+| String        | [xs:string][xml-primitives] | |
+| Binary        | [xs:base64Binary][xml-primitives] | |
+| URI           | [xs:anyURI][xml-primitives] following [RFC 3986][rfc3986]| |
+| URI-reference | [xs:anyURI][xml-primitives] following [RFC 3986][rfc3986] | |
+| Timestamp     | [xs:dateTime][xml-primitives] following [RFC 3339][rfc3339] | |
+
+Each context attribute is represented as an XML element whose name MUST exactly match
+that of a REQUIRED or OPTIONAL CloudEvent attribute, extension attribute names MUST
+adhere to, and align with, the conventions in defined in the [cloud event specification][ce-spec].
+
+See the [envelope](#4-envelope) for special handing of the `specversion` context attribute.
+
+Extension attributes SHOULD be expressed with the appropriate `xsi:type` to allow them
+to be exchanged without loss of type information. Extension attributes with no `xsi:type`
+discriminator SHOULD be interpreted with an implied type of `xs:string`
+
+``` xml
+    ...
+    <id>AAABBBCCCNNN0000</id>
+    ..
+    <time>2021-08-14T14:30:22-08:00</time>
+    ..
+    <myextension xsi:type="xs:string">my extension value</myextension>
+    <myboolean xsi:type="xs:boolean">false</myboolean>
+```
+
+## 3. Data
+
+The `Data` portion of a CloudEvent follows a similar model to that employed by
+the [JSON Format specification][json-format]. A `data` element is used to
+encapsulate the payload and an `xsi:type` used to discrimate the payload
+type.
+
+### 3.1 Binary Data
+
+MUST be carried in an element with an defined type of `xs:base64Binary`
+
+``` xml
+<data xsi:type="xs:base64Binary">.........</data>
+```
+
+### 3.2 Text Data
+
+MUST be carried in an element with an defined type of `xs:string`
+
+``` xml
+<data xsi:type="xs:string">This is text</data>
+```
+
+### 3.3 XML Data
+
+XML data MUST be carried in an element with a defined type of `xs:any`
+allowing a single XML element (with any required namespace definitions)
+to be represented.
+
+``` xml
+<data xsi:type="xs:any">
+    <myData>
+        ....
+    </myData>  
+</data>
+```
+
+## 4. Envelope
+
+Each CloudEvent is wholly represented as an XML element `<Event>` that
```

Comment:
I don't *think* we discussed the casing part here yesterday. It looks a little odd for all elements other than `Event` and `Batch` to be lowercase. (The casing part got lost in other aspects of the naming discussion.)

Given recent discussions around other specs, I'd suggest that using `<event>` and `<batch>` would cause less surprise to users.